src/components/blog/heading-with-anchor.tsx

```
@@ -0,0 +1,56 @@
+
+  // Generate ID from text if not provided
+  const headingId =
+    id ||
+    (typeof children === "string"
+      ? children
+          .toLowerCase()
+          .trim()
+          .replace(/\s+/g, "-")
+          .replace(/[^a-z0-9-_]/g, "")
+          .replace(/^-+|-+$/g, "") // Remove leading/trailing hyphens
+      : `heading-${Math.random().toString(36).substr(2, 9)}`); // Fallback for non-string
```

ESLint

[error] Error: Cannot call impure function during render

`Math.random` is an impure function. Calling an impure function can produce unstable results that update unpredictably when the component happens to re-render. (https://react.dev/reference/rules/components-and-hooks-must-be-pure#components-and-hooks-must-be-idempotent).

/home/jailuser/git/src/components/blog/heading-with-anchor.tsx:25:20
  23 |           .replace(/[^a-z0-9-_]/g, "")
  24 |           .replace(/^-+|-+$/g, "") // Remove leading/trailing hyphens
> 25 |       : `heading-${Math.random().toString(36).substr(2, 9)}`); // Fallback for non-string
     |                    ^^^^^^^^^^^^^ Cannot call impure function
  26 |
  27 |   const handleCopyLink = async () => {
  28 |     const url = `${window.location.origin}${window.location.pathname}#${headingId}`;

(react-hooks/purity)